auth-web/src/views/auth/staff/ContinuationAuthorizationReview.vue

```diff
@@ -115,19 +177,62 @@ export default class ContinuationAuthorizationReview extends Vue {
 
     if (!review || !results || !filing) {
       // eslint-disable-next-line no-console
-      console.log('Missing data for Continuation Review =', this.continuationReview)
+      console.log(`Missing data for Continuation Review ${this.reviewId} = ${this.continuationReview}`)
 
+      // show error dialog
       this.dialogTitle = 'Unable to fetch review'
       this.dialogText = 'An error occurred while fetching the review. Please try again.'
       this.$refs.errorDialog.open()
     }
   }
 
-  /** Prevents route changes if there's unsaved data. */
-  // *** TODO: move this to review result sub-component and finish implementation?
-  async beforeRouteLeave (to, from, next) {
-    const haveUnsavedChanges = false
-    if (haveUnsavedChanges) {
+  @Watch('reviewResult')
+  @Watch('emailBodyText')
+  onDataChanged (): void {
+    if (this.reviewResult || this.emailBodyText) {
+      this.haveUnsavedChanges = true
+    }
+  }
+
+  onClickCancel (): void {
+    this.haveUnsavedChanges = false
+    this.$router.push(Pages.STAFF_DASHBOARD)
+  }
+
+  async onClickSubmit (): Promise<void> {
+    if (!this.$refs.reviewResult.validate()) return
+
+    try {
+      this.isSubmitting = true
+      await BusinessService.saveContinuationReviewResult(this.reviewResult, this.emailBodyText)
+      this.haveUnsavedChanges = false
+
+      // indicate success via toast (snackbar) on main page
+      EventBus.$emit('show-toast', {
+        message: 'Review submitted successfully',
+        type: 'primary',
+        timeout: 3000
+      })
+    } catch (error) {
+      // eslint-disable-line no-console
+      console.log(`Error submitting review = ${error}`)
+
+      // show error dialog
+      this.dialogTitle = 'Unable to submit review'
+      this.dialogText = 'An error occurred while submitting the review. Please try again.'
+      this.$refs.errorDialog.open()
+    } finally {
+      this.isSubmitting = false
+    }
+
+    // if submit succeeded, route back to staff dashboard
+    if (!this.haveUnsavedChanges) this.$router.push(Pages.STAFF_DASHBOARD)
+  }
+
+  /** Intercepts route change (but not page unloads). */
```

Comment:
I can't see to connect to the `onpageunload` window event, probably due to misc event handlers in Auth Web. Oh well.